pytorch_lightning/callbacks/rich_model_summary.py

```diff
@@ -0,0 +1,79 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List, Union
+
+from pytorch_lightning.callbacks import ModelSummary
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _RICH_AVAILABLE
+from pytorch_lightning.utilities.model_summary import get_human_readable_count
+
+if _RICH_AVAILABLE:
+    from rich.console import Console
+    from rich.table import Table
+
+
+class RichModelSummary(ModelSummary):
```

Comment:
don't forget to add this file to the mypy list

Reply:
and let's add a docstring here? that this requires rich and instructions how to install? also maybe point to rich progressbar.

Reply:
Addressed!